membership-attribute-service/app/services/IdentityAuthService.scala

```diff
@@ -1,41 +1,55 @@
 import com.gu.identity.play.IdentityPlayAuthService.UserCredentialsMissingError
 import com.gu.monitoring.SafeLogger
-import com.gu.monitoring.SafeLogger._
+import models.{AccessClaims, AccessClaimsParser}
+import org.http4s.Uri
+
 import scala.concurrent.{ExecutionContext, Future}
 
-class IdentityAuthService(apiConfig: IdapiConfig)(implicit ec: ExecutionContext) extends AuthenticationService {
+class IdentityAuthService(apiConfig: IdapiConfig, oktaTokenVerifierConfig: OktaTokenVerifierConfig)(implicit ec: ExecutionContext) extends AuthenticationService {
 
   val idApiUrl = Uri.unsafeFromString(apiConfig.url)
 
-  val identityPlayAuthService = IdentityPlayAuthService.unsafeInit(idApiUrl, apiConfig.token, Some("membership"))
+  val identityPlayAuthService = {
+    val idapiConfig = IdapiAuthConfig(idApiUrl, apiConfig.token, Some("membership"))
+    IdentityPlayAuthService.unsafeInit(AccessClaimsParser,idapiConfig, oktaTokenVerifierConfig)
+  }
 
-  def user(implicit requestHeader: RequestHeader): Future[Option[User]] = {
+  def user(implicit requestHeader: RequestHeader): Future[Option[AccessClaims]] = {
     getUser(requestHeader)
-      .map(user => Option(user))
       .handleError { err =>
-        if (err.isInstanceOf[UserCredentialsMissingError])
-          // IdentityPlayAuthService throws an error if there is no SC_GU_U cookie or crypto auth token
-          // frontend decides to make a request based on the existence of a GU_U cookie, so this case is expected.
-          SafeLogger.info(s"unable to authorize user - no token or cookie provided")
-        else
-          SafeLogger.warn(s"valid request but expired token or cookie so user must log in again - $err")
-
+        err match {
+          case UserCredentialsMissingError(_) =>
+            // IdentityPlayAuthService throws an error if there is no SC_GU_U cookie or crypto auth token
+            // frontend decides to make a request based on the existence of a GU_U cookie, so this case is expected.
+            SafeLogger.info(s"unable to authorize user - no token or cookie provided")
+
+           case OktaVerifierException(validationError: ValidationError) =>
+            validationError match {
+              case OktaValidationError(originalException) =>
+                SafeLogger.warn(s"could not validate okta token - $validationError", originalException)
+              case _ =>
+                SafeLogger.warn(s"could not validate okta token - $validationError")
+             }
+          case err =>
+            SafeLogger.warn(s"valid request but expired token or cookie so user must log in again - $err")
+        }
         None
       }
   }
 
-  private def getUser(requestHeader: RequestHeader): Future[User] =
-    identityPlayAuthService
-      .getUserFromRequest(requestHeader)
-      .map { case (_, user) => user }
-      .unsafeToFuture()
-
+  private def getUser(requestHeader: RequestHeader): Future[Option[AccessClaims]] = {
```

Comment:
Will need scopes to be sent through from client app:
```suggestion
  private def getUser(requestHeader: RequestHeader, requiredScopes: List[String] = List.empty): Future[Option[AccessClaims]] = {
```